### ISSUE TITLE
Allow installing multiple versions of a package

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#837]
Yarn supports this: https://twitter.com/sebmck/status/873958247304232961

---

**Use case**

The use case for me is supporting multiple versions of packages in plugins. For example you could write a plugin that would support react-router@3 and @4 in the same module.

Atom's [service hub](https://github.com/atom/service-hub) is a good use case: "A semantically versioned provider/consumer system for global application services."

```
# Provide a 1.0 service
global.services.provide "order-pizza", "1.0.0",
  placeOrder: (size, topping) -> # ...

# Providing a 2.0 service with new API
placeOrder = ({size, toppings}) -> # ...
global.services.provide "order-pizza", "2.0.0", {placeOrder}

# Shimming the old API
global.services.provide "order-pizza", "1.0.0",
  placeOrder: (size, topping) ->
    placeOrder({size, toppings: [topping]})
```

So imagine you have plugins for `react`, such as `react-router`, `redux`, `react-router-redux`, etc.

When you use a version of `react-router` that works for `react@16`, you would need to increase to `react@16`, and then every other react-related module.

But if multiple versions were allowed, you would allow the user to always update to the latest version of a module, and then you could redirect them to the correct api.

Related #117 #838

**Workaround**

The current workaround is to wrap the different versions of the packages as separate packages e.g. `my-plugin-react-router` would depend on `my-plugin-react-router-3` (depends on `react-router@^3`) and `my-plugin-react-router-4` (depends on `react-router@^4`). You could then redirect requires/imports using webpack/require-hack to the appropriate version.

Worrying about updating to new versions of react, etc. would become a thing of the past, and you could just stay on the latest.

---

For now I am using the `@version` org to allow installing multiple deps. E.g.`@version/react-router-v3`. Need to alias/require-hack peer deps though...

**Comment by zkochan** [comment]
If possible, I'd implement it as an installation hook.

**Comment by zkochan** [comment]
What about the bins? I just tried the following with Yarn: `yarn add eslint3@npm:eslint@3 eslint4@npm:eslint@4` and it created one bin file only - eslint. This seems to right, I'd expect two: eslint3 and eslint4

**Comment by zkochan** [comment]
Published in [v1.23.0](https://github.com/pnpm/pnpm/releases/tag/v1.23.0)
